### DISCUSSION TITLE
How to Cleanup invalid jobs

### DISCUSSION THREAD

**Discussion by xantari** [quartznet/quartznet#1877] [Q&A] [answered]
Hi,

If an "IJob" implementation disappears and leaves entries in the persistent store (such as a SQL Server database), what is the best way to clean all those invalid job entries upon application startup so they don't keep firing errors because the IJob implementation no longer exists?  

My guess is you would have to search for all IJob implementations using reflection, then compare that to all the job registrations in Quartz and eliminate them if they don't exist but I'm not sure how to search by registered IJob type in Quartz .NET

Thanks!

**Comment by lahma** [comment, marked as answer]
Because you are removing the implementations, you should also keep track of removal. So one should remove such jobs from persistent store before removing them from binaries.

You could re-introduce the types that are missing and use the API the clean up (remove jobs and triggers) or as last resort clean up using SQL queries (better make a backup).
